smwe_rom/src/snes_utils/rom.rs

```diff
@@ -0,0 +1,74 @@
+use nom::{error::Error, IResult};
+
+use crate::{
+    error::{ParseErr, RomError},
+    snes_utils::{
+        addr::{Addr, AddrPc},
+        rom_slice::*,
+    },
+};
+
+pub const SMC_HEADER_SIZE: usize = 0x200;
+
+pub struct Rom(Vec<u8>);
+
+impl Rom {
+    pub fn new(mut data: Vec<u8>) -> Result<Self, RomError> {
+        let size = data.len() % 0x400;
+        if size == SMC_HEADER_SIZE {
+            Ok(Self(data.drain(SMC_HEADER_SIZE..).collect()))
+        } else if size == 0 {
+            Ok(Self(data))
```

Comment:
Seems like an empty rom should be an error.

Reply:
Ok, so `size == 0` doesn't mean the ROM is empty, it means that its size is a multiple of 1kB. If it is, it means the ROM file doesn't contain a SMC header at the start, which is just a padding of 512 bytes.

I should probably rename `size` to `mod_1k` and also add an error case for `data.len() == 0`.